server/bundler/extensions-loader.js

```diff
@@ -0,0 +1,34 @@
+const existsSync = require( 'fs' ).existsSync;
+const path = require( 'path' );
+const camelCase = require( 'lodash' ).camelCase;
+
+function getReducers( extensions = {} ) {
+	return extensions
+		.filter( ( extension ) => existsSync( path.join( 'client', 'extensions', extension, 'state', 'reducer.js' ) ) )
+		.map( ( extension ) => `'${ camelCase( extension ) }': require( 'extensions/${ extension }/state/reducer' )\n` );
+}
+
+function getExtensionsModule( extensions ) {
+	return `module.exports = {
+		reducers: function() {
+			return {
+				${ getReducers( extensions ) }
+			};
+		}
+	};
+`;
+}
+
+module.exports = function( content ) {
+	this.cacheable && this.cacheable();
+
+	const extensions = require( this.resourcePath );
```

Comment:
>where did these `this.` values come from?

These are provided by Webpack to loaders.

https://webpack.js.org/api/loaders/#the-loader-context